lms/djangoapps/courseware/access.py

```diff
@@ -279,9 +279,27 @@ def can_view_courseware_with_prerequisites():  # pylint: disable=invalid-name
         else:
             return True
 
+    def can_view_courseware_with_entrance_exam():  # pylint: disable=invalid-name
+        """
+        Checks if entrance exam feature is enabled and course has entrance exam
+        and user is neither staff nor anonymous and user is not allowed to skip
+        entrance exam then it returns False if user has not passed entrance exam
+        otherwise return True.
+        """
+        if settings.FEATURES.get('ENTRANCE_EXAMS') \
+                and getattr(course, 'entrance_exam_enabled', False) \
+                and not _has_staff_access_to_descriptor(user, course, course.id) \
+                and not user.is_anonymous() \
```

Comment:
Thanks for pointing it out. I'm returning entrance exam permission will fail in case of anonymous user now.